### PR TITLE
Add more config and extract async tasks to function

### PR DESCRIPTION
**Pull request by saying121**
Add heartbeat time `duration` and `enable` field for rpc config.

### CODE REVIEW

File: Cargo.toml
Comment (Thaumy) [inline Cargo.toml:1]: Updating dependencies doesn't seem relevant to the functionality of this commit, it would be better to separate it out.

File: src/services/config.rs [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 }
 
-impl RpcConfig {
-    pub fn addr(&self) -> &str {
-        &self.addr
-    }
-
-    pub fn token(&self) -> &str {
-        &self.token
+impl Default for RpcConfig {
```
Comment (Thaumy) [inline src/services/config.rs:11]: The `Default` impl is useless because it will never establish a valid RPC connection.
Rust docs defines `Default` as "A trait for giving a type a useful default value", see: https://doc.rust-lang.org/std/default/trait.Default.html

File: src/services/rpc.rs [hunk 2]
```diff
@@ -19,20 +21,24 @@ use crate::services::{
...
     client: PshServiceClient<tonic::transport::Channel>,
     raw_info: RawInfo,
+    duration: Duration,
 }
 
 impl RpcClient {
-    pub async fn new(addr: &str, token: String) -> Result<Self> {
+    pub async fn new(config: RpcConfig) -> Result<Self> {
         let client: PshServiceClient<tonic::transport::Channel> =
-            PshServiceClient::connect(format!("https://{}", addr)).await?;
```
Comment (Thaumy) [inline src/services/rpc.rs:32]: This change should be in a separate commit since it's not relevant to your commit message.

File: src/config/mod.rs [hunk 3]
```diff
@@ -156,8 +156,10 @@ stderr_file = "/tmp/psh.stderr"
 working_directory = "/"
 
 [rpc]
-addr = "www.optimatist.com"
+enable = true
```
Comment (Thaumy) [inline src/config/mod.rs:159]: It's a good idea to split "make rpc switchable" and "add heartbeat time" into two commits cuz it makes your commit history cleaner and easier to review.

File: src/main.rs [hunk 4]
```diff
@@ -108,7 +79,37 @@ fn main() -> anyhow::Result<()> {
 
     engine.run(&component_args[0])?;
 
-    let _ = hd.join();
+    let _ = async_handle.join();
 
     Ok(())
 }
+
+fn async_tasks(rpc_conf: RpcConfig, otlp_conf: OtlpConfig) -> JoinHandle<Result<()>> {
```
Comment (Thaumy) [inline src/main.rs:87]: The commit message "extract async_tasks to a function" makes me confused.
Maybe you meant "extract async tasks to `async_tasks`"?

File: src/main.rs [hunk 5]
```diff
@@ -37,7 +35,10 @@ use opentelemetry_otlp::ExportConfig;
 use runtime::PshEngineBuilder;
 use utils::check_root_privilege;
 
-fn main() -> anyhow::Result<()> {
```
Comment (Thaumy) [inline src/main.rs:40]: Turning `anyhow::Result<()>` to `Result<()>` just changes the code style.
A separate commit with message "import `anyhow::Result`" is appropriate to it.

**Review by Thaumy — CHANGES_REQUESTED** [review]

File: src/services/config.rs (hunk 1 above)
Comment (saying121) [reply src/services/config.rs:11]: It's for generate a default `config.toml` file

Comment (Thaumy) [reply src/services/config.rs:11]: It's heavy to impl `Default` just for generating the default `config.toml` file.

Comment (Thaumy) [reply src/services/config.rs:11]: I will merge this since there is already a lot of `Default` impls just for docs gen. It might be better to add configuration templates in the future.

**Review by Thaumy — APPROVED** [review]
